src/jit/decomposelongs.cpp

```diff
@@ -68,21 +68,39 @@ void DecomposeLongs::DecomposeBlock(BasicBlock* block)
     assert(block == m_compiler->compCurBB); // compCurBB must already be set.
     assert(block->isEmpty() || block->IsLIR());
 
-    m_block = block;
+    m_blockWeight = block->getBBWeight(m_compiler);
+    m_range = &LIR::AsRange(block);
+    DecomposeRange();
+}
+
+void DecomposeLongs::DecomposeRange(Compiler* compiler, unsigned blockWeight, LIR::Range& range)
+{
```

Comment:
These two methods don't have a block comment describing them.
Also in general I don't like the names of methods to be  overloaded for no good reason.
(Such methods are a bit harder to debug using windbg as you can't easily set a breakpoint in them by name)
Also I think that you could combine these two into a single method by changing the callsite above.